tuplex/adapters/cpython/src/PythonSerializer.cc

```diff
@@ -129,7 +129,15 @@ namespace tuplex {
 
                 int curr_index = curr.back();
                 python::Type current_type = tree.fieldType(curr);
-                PyObject *elem_to_insert = createPyObjectFromMemory(ptr + current_buffer_index, current_type, bitmap, bitmap_index);
+                PyObject *elem_to_insert = nullptr;
+                if (current_type.isOptionType() && current_type.getReturnType().isTupleType()) {
```

Comment:
can you add a comment on what the logic here is? I.e., why is the case Opt[Tuple[....]] so special?